sky/backends/cloud_vm_ray_backend.py

```diff
@@ -2434,29 +2436,42 @@ def teardown_no_lock(self,
                         f'gcloud compute tpus tpu-vm list --filter='
                         f'\\(labels.ray-cluster-name={cluster_name}\\) '
                         f'--zone={zone} --format=value\\(name\\)')
-                    terminate_cmd = (
-                        f'gcloud compute tpus tpu-vm delete --zone={zone}'
-                        f' --quiet $({query_cmd})')
+                    returncode, stdout, stderr = log_lib.run_with_log(
+                        query_cmd,
+                        log_abs_path,
+                        shell=True,
+                        stream_logs=False,
+                        require_outputs=True)
+
+                    # Needs to create a list as GCP does not allow deleting
+                    # multiple TPU VMs at once
```

Comment:
Just a question: does this mean we cannot parallelize the termination loop in L2468?

Reply:
We can parallelize the loop which will send multiple http requests at the same time. Do you think we should do that?

Reply:
No, I was just curious. We can do that later once we have real user complaints.